File: frontend/src/api/training.ts
/**
 * Training API - Backend communication for training data management
 */

import apiClient from './client'
import type {
  TrainRequest,
  TrainResponse,
  GetTrainingDataResponse,
  RemoveTrainingDataResponse,
} from '@/types/api'

export const trainingAPI = {
  /**
   * Get all training data
   */
  async getTrainingData(): Promise<GetTrainingDataResponse> {
    const { data } = await apiClient.get<GetTrainingDataResponse>('/api/v0/training')
    return data
  },

  /**
   * Add training data (DDL, documentation, or Q&A pair)
   */
  async addTraining(request: TrainRequest): Promise<TrainResponse> {
    const { data } = await apiClient.post<TrainResponse>('/api/v0/training', request)
    return data
  },

  /**
   * Remove training data by ID
   */
  async removeTraining(id: string): Promise<RemoveTrainingDataResponse> {
    const { data } = await apiClient.delete<RemoveTrainingDataResponse>(
      `/api/v0/training/${id}`
    )
    return data
  },
}

export default trainingAPI

File: backend/app/models/training.py
"""
Training data management models.
"""

from pydantic import BaseModel, Field
from typing import Optional, Dict, Any, List


class TrainRequest(BaseModel):
    """Request to add training data."""
    ddl: Optional[str] = Field(None, description="DDL statement")
    documentation: Optional[str] = Field(None, description="Table/column documentation")
    question: Optional[str] = Field(None, description="Example question")
    sql: Optional[str] = Field(None, description="Example SQL query")


class TrainResponse(BaseModel):
    """Response after adding training data."""
    status: str
    message: str


class GetTrainingDataResponse(BaseModel):
    """Response with all training data."""
    training_data: List[Dict[str, Any]]
    count: int


class RemoveTrainingDataRequest(BaseModel):
    """Request to remove training data."""
    id: str = Field(..., description="Training data ID to remove")


class RemoveTrainingDataResponse(BaseModel):
    """Response after removing training data."""
    status: str
    message: str
